nncf/experimental/onnx/algorithms/quantizer_range_finder.py

```diff
@@ -0,0 +1,151 @@
+"""
+ Copyright (c) 2022 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from nncf.common.hardware.config import HWConfigType
+from nncf.common.hardware.config import HW_CONFIG_TYPE_TARGET_DEVICE_MAP
+from nncf.common.quantization.quantizer_propagation.solver import QuantizerPropagationSolver
+from nncf.common.quantization.structs import QuantizableWeightedLayerNode
+from nncf.common.quantization.structs import QuantizerConfig
+from nncf.common.quantization.structs import QuantizationMode
+from nncf.common.insertion_point_graph import InsertionPointGraph
+
+from nncf.experimental.post_training.algorithms.quantizer_range_finder import QuantizerRangeFinderAlgorithm
+from nncf.experimental.post_training.compressed_model import CompressedModel
+from nncf.experimental.onnx.graph.transformations.layout import ONNXTransformationLayout
+from nncf.experimental.onnx.graph.metatypes.onnx_ops import GENERAL_WEIGHT_LAYER_METATYPES
+from nncf.experimental.onnx.algorithms.quantization.default_quantization import DEFAULT_ONNX_QUANT_TRAIT_TO_OP_DICT
+from nncf.experimental.onnx.graph.transformations.commands import ONNXQuantizerInsertionCommand
+from nncf.experimental.onnx.engine import ONNXEngine
+from nncf.experimental.post_training.algorithms.quantizer_range_finder import QuantizerRangeFinderParameters
+from nncf.experimental.onnx.statistics.collectors import ONNXMinMaxStatisticCollector
+from nncf.experimental.onnx.graph.model_transformer import ONNXModelTransformer
+
+from nncf.experimental.onnx.hardware.fused_patterns import ONNX_HW_FUSED_PATTERNS
+
+from nncf.experimental.onnx.algorithms.quantization.helper import calculate_activation_quantizer_parameters
+from nncf.experimental.onnx.algorithms.quantization.helper import calculate_weight_quantizer_parameters
+
+from nncf.experimental.onnx.hardware.config import ONNXHWConfig
+
+QUANTIZATION_LAYER_METATYPES = GENERAL_WEIGHT_LAYER_METATYPES
+
+
+class ONNXQuantizerRangeFinderAlgorithm(QuantizerRangeFinderAlgorithm):
```

Comment:
It doesn't seem that this algo only finds quantizer ranges. It also seems to do end2end quantization, so either the name of the class is incorrect, or it does something that it shouldn't.

Reply:
Agreed. Change the name from ONNXQuantizerRangeFinderAlgorithm to ONNXMinMaxQuantization